src/BuildScriptGenerator/DotNetCore/VersionProviders/DotNetCoreExternalVersionProvider.cs

```diff
@@ -0,0 +1,112 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.XPath;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator.Common;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.DotNetCore
+{
+    public class DotNetCoreExternalVersionProvider : ExternalSdkStorageVersionProviderBase, IDotNetCoreVersionProvider
+    {
+        private readonly BuildScriptGeneratorOptions commonOptions;
+        private readonly IExternalSdkProvider externalSdkProvider;
+        private Dictionary<string, string> versionMap;
+        private string defaultRuntimeVersion;
+
+        public DotNetCoreExternalVersionProvider(
+            IOptions<BuildScriptGeneratorOptions> commonOptions,
+            IExternalSdkProvider externalSdkProvider,
+            ILoggerFactory loggerFactory)
+            : base(commonOptions, externalSdkProvider, loggerFactory)
+        {
+            this.commonOptions = commonOptions.Value;
+            this.externalSdkProvider = externalSdkProvider;
+        }
+
+        public Dictionary<string, string> SupportedVersionsMap { get; }
+
+        public string GetDefaultRuntimeVersion()
+        {
+            this.GetVersionInfo();
+            return this.defaultRuntimeVersion;
+        }
+
+        public Dictionary<string, string> GetSupportedVersions()
+        {
+            this.GetVersionInfo();
+            return this.versionMap;
+        }
+
+        /// <summary>
+        /// Gets list of all blobs for dotnet platform and determines the supported and default versions.
+        /// -----------
+        /// This works slightly differently than <see cref="ExternalSdkStorageVersionProviderBase.GetAvailableVersionsFromExternalProvider"/>,
+        /// as the dotnet supported versions are a mapping of runtime version -> sdk version. This means that we need to find
+        /// both runtime version and sdk version metadata associated with each file.
+        /// </summary>
+        public void GetVersionInfo()
```

Comment:
We can move this whole method into ExternalSdkStorageVersionProviderBase

Reply:
We can then have a variable initialized in constructor of this class which is used by the method in base class to convey dotnet platform